### PR TITLE
fix: AM-327 support for null relationship

### PR DESCRIPTION
**Pull request by abhijitdiwan**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-327

### Change description ###
fix: Support Null Relationship (AM-327)

### CODE REVIEW

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java [hunk 1]
```diff
@@ -111,4 +111,12 @@ private ExplicitAccessGrant createPermissionsForResourceForMultipleUsers(String
             .relationship(roleName)
             .build();
     }
+
+    @Test
+    void whenCreatingResourceAccessResourceAccessAppearsInDatabaseWIthNullRelationship() {
```
Comment (shashank-rastogi) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java]: the name of the method does not make sense

Moreover, it is not clear what this test case is trying to achieve?

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/helpers/DatabaseHelperRepository.java [hunk 2]
```diff
@@ -59,6 +59,10 @@
         + "where resource_id = :resourceId")
     int countExplicitPermissions(String resourceId);
 
+    @SqlQuery("select relationship from access_management "
+        + "where resource_id = :resourceId")
+    String getNullRelationShipForAnnotation(String resourceId);
+
     @SqlQuery("select * from access_management "
         + "where resource_id = :resourceId")
```
Comment (shashank-rastogi) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/helpers/DatabaseHelperRepository.java]: there is a remote possibility that resource id may repeat across different services/resource types

Therefore, should be handled appropriately

**Review by shashank-rastogi — CHANGES_REQUESTED** [review]

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java (hunk 1 above)
Comment (abhijitdiwan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java]: It's more specific to check null relationship/role. And check  Explicit Access record gets created or not with null relationship.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/helpers/DatabaseHelperRepository.java (hunk 2 above)
Comment (abhijitdiwan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/helpers/DatabaseHelperRepository.java]: right may be we can use existing method findExplicitPermissions method. Will use that

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java (hunk 1 above)
Comment (abhijitdiwan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java]: will rename this

**Review by shashank-rastogi — APPROVED** [review]

**Comment by abhijitdiwan** [comment]
AM 327 : Check null relationship for Annotations

**Pull request merged**